pkg/cache/cache.go

```diff
@@ -228,6 +228,9 @@ func (c *Cache) updateClusterQueues() sets.Set[kueue.ClusterQueueReference] {
 		curStatus := cq.Status
 		if prevStatus == pending && curStatus == active {
 			cqs.Insert(cq.Name)
+			for _, w := range cq.Workloads {
```

Comment:
Is this map guaranteed to be correctly populated here or is it raceish? Maybe it'd be better to list the workloads from the client's cache which hopefully should have synced before the Create/Reconcile events happen

Reply:
Well, I think it is not guaranteed at the moment indeed, which is partly why the tests are still failing. I think this issue is deeper than TAS actually. I will investigate, maybe indeed we should relist from informer, assuming that WaitForCacheSync happened.

Reply:
Ok, my current thinking is to keep the CQ as inactive as long as there are any TAS workloads, which are not accounted due to nil TAS cache. So we would keep the map of workloads deferredTAS name to Workload, and keep the CQ inactive as long as not empty. I will try to have a prototype  tomorrow

Reply:
I have updated the approach. PTAL.